pkg/types/openstack/validation/realvalidvaluesfetcher.go

```diff
@@ -21,13 +24,22 @@ func NewValidValuesFetcher() ValidValuesFetcher {
 
 // GetCloudNames gets the valid cloud names. These are read from clouds.yaml.
 func (f realValidValuesFetcher) GetCloudNames() ([]string, error) {
-	clouds, err := clientconfig.LoadCloudsYAML()
+	filename, content, err := clientconfig.FindAndReadCloudsYAML()
 	if err != nil {
 		return nil, err
 	}
+
+	var clouds clientconfig.Clouds
+	err = yaml.Unmarshal(content, &clouds)
+	if err != nil {
+		return nil, fmt.Errorf("failed to unmarshal yaml: %v", err)
+	}
+
+	logrus.Info("Credentials loaded from file ", filename)
+
 	i := 0
-	cloudNames := make([]string, len(clouds))
-	for k := range clouds {
+	cloudNames := make([]string, len(clouds.Clouds))
+	for k := range clouds.Clouds {
 		cloudNames[i] = k
 		i++
 	}
```

Comment:
I don't think this needs to be changed...

a more preferred approach might be to setup realValidValuesFetcher the session using the GetSession instead of hosting this similar method in two places.

Reply:
As per @wking I moved `realValidValuesFetcher` to `pkg/asset` ([precedent](https://github.com/openshift/installer/blob/master/pkg/asset/installconfig/aws/validation.go) in AWS validations). It's making more sense now and able to reuse all logic. PTAL